includes/classes/Feature/VectorEmbeddings/VectorEmbeddings.php

```diff
@@ -0,0 +1,460 @@
+		$this->slug = 'vector_embeddings';
+
+		$this->title = esc_html__( 'Vector Embeddings', 'elasticpress-labs' );
+
+		$this->requires_install_reindex = true;
+
+		$this->summary = __(
+			'This feature enables storage of vector embeddings, a numerical representation of the indexed content that can capture semantic relationships and similarities between data points. These embeddings are often used by AI models to process and understand complex information more efficiently and are used for features like natural language processing, recommendations and computer vision.',
+			'elasticpress-labs'
+		);
+
+		parent::__construct();
+	}
+
+	/**
+	 * Connects the Module with WordPress using Hooks and/or Filters.
+	 *
+	 * @return void
+	 */
+	public function setup() {
+		$this->indexables['post'] = new Indexables\Post( $this );
+		$this->indexables['post']->setup();
+	}
+
+	/**
+	 * Tell user whether requirements for feature are met or not.
+	 *
+	 * @return FeatureRequirementsStatus Requirements object
+	 */
+	public function requirements_status() {
+		$status = new \ElasticPress\FeatureRequirementsStatus( 1 );
+
+		// Vector support was added in Elasticsearch 7.0.
+		if ( version_compare( Elasticsearch::factory()->get_elasticsearch_version(), '7.0', '<=' ) ) {
+			$status->code    = 2;
+			$status->message = esc_html__( 'You need to have Elasticsearch with version >7.0.', 'elasticpress-labs' );
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Set the `settings_schema` attribute
+	 */
+	public function set_settings_schema() {
+		$this->settings_schema = [
+			[
+				'key'   => 'ep_embeddings_api_key',
+				'label' => __( 'OpenAI API Key', 'elasticpress-labs' ),
+				'help'  => sprintf(
+					wp_kses(
+						/* translators: %1$s: OpenAI sign up URL */
+						__( 'Don\'t have an OpenAI account yet? <a title="Sign up for an OpenAI account" href="%1$s">Sign up for one</a> in order to get your API key.', 'elasticpress-labs' ),
+						[
+							'a' => [
+								'href'  => [],
+								'title' => [],
+							],
+						]
+					),
+					esc_url( 'https://platform.openai.com/signup' )
+				),
+				'type'  => 'text',
+			],
+			[
+				'help'    => __( 'OpenAI Embeddings API Url', 'elasticpress-labs' ),
+				'key'     => 'ep_embeddings_api_url',
+				'label'   => __( 'OpenAI Embeddings API Url', 'elasticpress-labs' ),
+				'type'    => 'text',
+				'default' => $this->default_settings['ep_embeddings_api_url'],
+			],
+			[
+				'help'    => __( 'OpenAI Embedding model', 'elasticpress-labs' ),
+				'key'     => 'ep_embeddings_embedding_model',
+				'label'   => __( 'The name of the embedding model to use', 'elasticpress-labs' ),
+				'type'    => 'text',
+				'default' => $this->default_settings['ep_embeddings_embedding_model'],
+			],
+			[
+				'help'    => __( 'Embedding model dimensions', 'elasticpress-labs' ),
+				'key'     => 'ep_embeddings_dimensions',
+				'label'   => __( 'The number of dimensions supported by your embedding model', 'elasticpress-labs' ),
```

Comment:
@felipeelia I'm getting a 4xx response from the features endpoint: `Invalid parameter(s): vector_embeddings`. This is also the case the with the RAG setting for Number of Posts - both of these are `number` types. I tried changing them to 'text' but it doesn't fix the issue. Saving works again if I comment-out this item of the schema. Same for the RAG item

Reply:
Hey @psorensen! This was already fixed by https://github.com/10up/ElasticPress/pull/4072 (not released yet.) Any chance you can test it again using ElasticPress's `develop` branch? Thanks!

Reply:
oh great! Disregard!